Supervision Work/src/uk/ac/cam/ap886/oopjava/supervisionProjects/Hangman/CLI.java

```diff
@@ -0,0 +1,209 @@
+package uk.ac.cam.ap886.oopjava.supervisionProjects.Hangman;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+public class CLI implements UserInterface {
+	
+	
+	Game game;
+	BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+	
+	
+	//----------------
+	
+	public CLI (Game newGame){
+		game = newGame;
+	}
+	
+	
+	
+	private void prn(String arg){
+		System.out.println(arg);
+	}
+	
+	private String scn(){
+		try {
+			return br.readLine().trim();
+		} catch (IOException e) {
+			e.printStackTrace();
+			return null;
+		}
+	}
+	
+	private char getch(){
+		try{
+			return br.readLine().charAt(0);
+		}catch(IOException e){
+			e.printStackTrace();
+			return '\0';
+		}
+	}
+
+	@Override
+	public String playerName() {
+		prn("Please Enter a Player's name");
+		return scn();
+	}
+	
+	public boolean parseYesNo(String input){
+		input=input.toLowerCase();
+		switch (input){
+		case "yes":
+		case "y":
+		case "u":
+		case "t":
+			return true;
+		default :
+			return false;
+		}
+	}
+	
+	private void drawHangman(int stage){
```

Comment:
Consider using an enum, this seems error prone. Or, if you do want to use int (which makes sense as you naturally count stages) throw an IllegalArgumentException if stage is not within the allowed range.